src/validation/async/index.ts

```diff
@@ -152,3 +156,57 @@ export const filingPeriodTrustStartDateValidations = async (req: Request) => is_
 
 export const filingPeriodTrustCeaseDateValidations = async (req: Request) => is_date_within_filing_period_trusts(req, historicalBOEndDateContext, ErrorMessages.CEASED_DATE_BEFORE_FILING_DATE);
 
+export const beneficialOwnersTypeEmptyNOCList = async (req: Request, appData: ApplicationData): Promise<ValidationError[]> => {
+  const allowedUrls = [
+    [config.REGISTER_AN_OVERSEAS_ENTITY_URL, config.BENEFICIAL_OWNER_TYPE_PAGE],
+    [config.UPDATE_AN_OVERSEAS_ENTITY_URL, config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE]
+  ];
+
+  const allowed: boolean = isAllowedUrls(allowedUrls, req);
+
+  const errors: ValidationError[] = [];
+
+  if (!allowed) {
+    return errors;
+  }
+
+  try {
+    const isActive = isActiveFeature(config.FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC);
+
+    const boiList = checkIfBeneficialOwnerHasNOC(appData?.[BeneficialOwnerIndividualKey] ?? [], isActive);
+    const booList = checkIfBeneficialOwnerHasNOC(appData?.[BeneficialOwnerOtherKey] ?? [], isActive);
+    const bogList = checkIfBeneficialOwnerHasNOC(appData?.[BeneficialOwnerGovKey] ?? [], isActive);
+
+    const boList: string[] = [...boiList, ...booList, ...bogList];
+
+    if (boList.length) {
+      throw new Error(`${ErrorMessages.MISSING_NATURE_OF_CONTROL} ${boList.join(", ")}`);
+    }
+
+    return errors;
+  } catch (error) {
+    errors.push({
+      value: '',
+      msg: error.message,
+      param: 'beneficial_owner_list',
+      location: 'body',
+    });
+
+    return errors;
+  }
+};
+
+const checkIfBeneficialOwnerHasNOC = (beneficialOwner, isActive: boolean): string[] => {
+  return beneficialOwner?.filter(bo =>
+    !bo?.beneficial_owner_nature_of_control_types?.length &&
```

Comment:
Formatting looks a bit odd on this line - only 4 spaces instead of 8, or is that GitHub?

Reply:
There is no formatting issue

Reply:
Line 201 has 4 spaces and lines 206 to 209 6. Is that intended/correct?